packages/components/src/components/segmented-control/segmented-control-item/segmented-control-item.tsx

```diff
@@ -66,17 +66,19 @@ export class SegmentedControlItem {
     validateProps(this, propTypes);
     // this additional validation is still needed because undefined is allowed with current propTypes
     throwIfPropIsUndefined(this.host, 'value', this.value);
+    const hasIcon = !!this.icon || !!this.iconSource;
 
     attachComponentCss(
       this.host,
       getComponentCss,
       this.disabled,
       this.host.selected,
-      this.host.backgroundColor || 'background-default', // default as fallback
+      hasIcon,
       this.host.theme || 'light' // default as fallback
     );
 
     const PrefixedTagNames = getPrefixedTagNames(this.host);
+    const iconColor = getIconColor(this.disabled);
```

Comment:
I would remove this and keep it inline since then it is only calculated when needed. Currently this is always called even when there is no icon..

Reply:
Fixed